src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -1364,6 +1580,387 @@ static void DrawMapSettings_Colors()
 			regenerate = true;
 	}
 
+	ImColor highColor = HighlightColor.ToImColor();
+	if (ImGui::ColorEdit3("Highlight", &highColor.Value.x))
+	{
+		HighlightColor.Blue = static_cast<uint8_t>(highColor.Value.z * 255);
+		HighlightColor.Green = static_cast<uint8_t>(highColor.Value.y * 255);
+		HighlightColor.Red = static_cast<uint8_t>(highColor.Value.x * 255);
+		HighlightColor.Alpha = 255;
+
+		WritePrivateProfileInt("Map Filters", "High-Color", HighlightColor.ToRGB(), INIFileName);
+		regenerate = true;
+	}
+
+	ImGui::Separator();
+
+	ImColor mapLocColor = gDefaultMapLocParams.color.ToImColor();
+	if (ImGui::ColorEdit3("MapLoc", &mapLocColor.Value.x))
+	{
+		if (!IsMapLocsStyleOverridden())
+		{
+			gOverrideMapLocParams.color.Blue = static_cast<uint8_t>(mapLocColor.Value.z * 255);
+			gOverrideMapLocParams.color.Green = static_cast<uint8_t>(mapLocColor.Value.y * 255);
+			gOverrideMapLocParams.color.Red = static_cast<uint8_t>(mapLocColor.Value.x * 255);
+			gOverrideMapLocParams.color.Alpha = 255;
+		}
+
+		gDefaultMapLocParams.color.Blue = static_cast<uint8_t>(mapLocColor.Value.z * 255);
+		gDefaultMapLocParams.color.Green = static_cast<uint8_t>(mapLocColor.Value.y * 255);
+		gDefaultMapLocParams.color.Red = static_cast<uint8_t>(mapLocColor.Value.x * 255);
+		gDefaultMapLocParams.color.Alpha = 255;
+
+		WritePrivateProfileInt("MapLoc", "Red", gDefaultMapLocParams.color.Red, INIFileName);
+		WritePrivateProfileInt("MapLoc", "Green", gDefaultMapLocParams.color.Green, INIFileName);
+		WritePrivateProfileInt("MapLoc", "Blue", gDefaultMapLocParams.color.Blue, INIFileName);
+		UpdateDefaultMapLocInstances();
+		regenerate = true;
+	}
+
+	ImGui::Separator();
+
+	ImColor radiusColor = gDefaultMapLocParams.circleColor.ToImColor();
+	if (ImGui::ColorEdit3("MapLoc Radius", &radiusColor.Value.x))
+	{
+		if (!IsMapLocsStyleOverridden())
+		{
+			gOverrideMapLocParams.circleColor.Blue = static_cast<uint8_t>(radiusColor.Value.z * 255);
+			gOverrideMapLocParams.circleColor.Green = static_cast<uint8_t>(radiusColor.Value.y * 255);
+			gOverrideMapLocParams.circleColor.Red = static_cast<uint8_t>(radiusColor.Value.x * 255);
+			gOverrideMapLocParams.circleColor.Alpha = 255;
+		}
+
+		gDefaultMapLocParams.circleColor.Blue = static_cast<uint8_t>(radiusColor.Value.z * 255);
+		gDefaultMapLocParams.circleColor.Green = static_cast<uint8_t>(radiusColor.Value.y * 255);
+		gDefaultMapLocParams.circleColor.Red = static_cast<uint8_t>(radiusColor.Value.x * 255);
+		gDefaultMapLocParams.circleColor.Alpha = 255;
```

Comment:
you're calculating the same color twice, in these sections.

Reply:
MQcolor has a constructor that takes ImColor.

Theres a lot of these, i'm not commenting on each.